### PR TITLE
Improve flatpak

### PR DESCRIPTION
**Pull request by kowalski7cc**
Improve the flatpak build for with the following points:

- Use a better image for the icon, from the macOS build
- Add a gitignore
- Use the correct xdg desktop category (https://specifications.freedesktop.org/menu-spec/latest/apas02.html)
- Build SDL2 with libdecor for Wayland support
- Remove X11 socket and add Wayland socket
- Patch upstream appstream

### CODE REVIEW

**Comment by kowalski7cc** [comment]
bot, build org.nongnu.enigma

File: org.nongnu.enigma.yml [hunk 1]
```diff
@@ -40,12 +122,19 @@ modules:
...
+      - type: git
+        url: https://github.com/apache/xerces-c
+        tag: v3.2.3
+        commit: cf1912ac95d4147be08aef4e78f894a3919277d9
+
+  - name: libpng
+    sources:
+      - type: git
+        url: https://github.com/glennrp/libpng
+        tag: v1.6.35
```
Comment (hfiguiere) [inline org.nongnu.enigma.yml]: does it really need to build libpng? it's in the runtime

File: org.nongnu.enigma.yml [hunk 2]
```diff
@@ -80,9 +169,7 @@ modules:
...
       - type: patch
         path: 0001-Fix-.desktop-to-success-the-desktop-filde-validate.patch
-      - type: file
-        path: org.nongnu.enigma.metainfo.xml
+      - type: patch
+        path: 0001-fix-appstream.patch
     post-install:
-      - install -d /app/share/icons/hicolor/128x128/apps/
-      - install -Dm644 --target-directory=${FLATPAK_DEST}/share/metainfo ${FLATPAK_ID}.metainfo.xml
-      - convert /app/share/icons/hicolor/48x48/apps/enigma.png -resize 128x128 /app/share/icons/hicolor/128x128/apps/enigma.png
```
Comment (hfiguiere) [inline org.nongnu.enigma.yml:88]: I think that's the reason of imagemagick in the package, isn't it?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.nongnu.enigma.yml (hunk 2 above)
Comment (kowalski7cc) [reply org.nongnu.enigma.yml:88]: Nope, I saw the enigma makefile doing stuff during build, unrelated to this step

File: org.nongnu.enigma.yml (hunk 1 above)
Comment (kowalski7cc) [reply org.nongnu.enigma.yml]: Probably then it could be removed, i will try tomorrow to build it locally without this and see if it works

**Comment by kowalski7cc** [comment]
After building and ~playing~ testing, I can confirm it works

**Review by hfiguiere — APPROVED** [review]

**Comment by kowalski7cc** [comment]
Nice :smile:

**Comment by kowalski7cc** [comment]
Hello, any news @MrTarantoga @hfiguiere?

**Comment by MrTarantoga** [comment]
I am sorry, I have missed the collaboration [invention](https://github.com/flathub/flathub/pull/2653#issuecomment-981389900). Can you invite me again @hfiguiere? That's also the case why I answer too late. Does not recognize that someone is improving the flatpak build.

**Comment by hfiguiere** [comment]
@barthalion can you resend the invite? Thanks.

**Comment by barthalion** [comment]
Hi @MrTarantoga, I just re-invited you to the repo. Please make sure to accept it so it doesn't expire again. Thanks!

**Pull request merged**